catalog/to-consul/syncer.go

```diff
@@ -267,6 +267,10 @@ func (s *ConsulSyncer) watchService(ctx context.Context, name, namespace string)
 			return err
 		}, backoff.WithContext(backoff.NewExponentialBackOff(), ctx))
 		if err != nil {
+			// If the context is ended, then we end.
+			if ctx.Err() != nil {
+				return
+			}
```

Comment:
The docs for this function say that the `Err()` returns non-nil only if Done is closed. So falling through to continue here would mean that on the next iteration of the loop, it'll fall to `ctx.Done()` case, right? Just want to understand what's happening better 😃

Reply:
Yeah exactly so it will go back but immediately exit (after making the api call) and so loop over and over.

Reply:
Oh wait, good catch yeah we don't need this here.

Reply:
Wouldn't it just hit a `return` statement in the ctx.Done case? So if we don't have the `if` statement and there is an error (i.e. context is done), it will:
1. call `continue`
1. enter the loop again
1. ctx is now done, so it will go to the `ctx.Done()` case, which calls `return`

I'm just curious if the if is necessary here if on the next iteration of the loop the function returns anyways. I could be misunderstanding this though!

Reply:
No you're correct. It doesn't need the early exit in this case.